src/service/viewport-impl.js

```diff
@@ -1414,10 +1414,14 @@ export class ViewportBindingIosEmbedWrapper_ {
   constructor(win) {
     /** @const {!Window} */
     this.win = win;
+    const topClasses = this.win.document.documentElement.className;
+    this.win.document.documentElement.className = '';
     this.win.document.documentElement.classList.add('-amp-ios-embed');
 
     /** @private @const {!Element} */
-    this.wrapper_ = this.win.document.createElement('i-amp-html-wrapper');
+    this.wrapper_ = this.win.document.createElement('html');
```

Comment:
Either way we have pretty horrible hacks and pretty horrible bugs. Through all the testing, however, this method appears to have fewer bugs and has a better performance. Among other things it allows header scrolling, flying carpet, correct scroll height calculation and many other benefits. Obviously, I can't wait for the day when we no longer need this.